react/src/views/Transfers/CreateTransferShipment/CreateTransferShipmentView.tsx

```diff
@@ -0,0 +1,178 @@
+import { useNotification } from "hooks/hooks";
+import APILoadingIndicator from "components/APILoadingIndicator";
+import { useNavigate, useParams } from "react-router-dom";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+
+import {
+  AllAcceptedTransferAgreementsQuery,
+  CreateTransferShipmentMutation,
+  CreateTransferShipmentMutationVariables,
+  TransferAgreementType,
+} from "types/generated/graphql";
+import { BASE_ORG_FIELDS_FRAGMENT, TRANSFER_AGREEMENT_FIELDS_FRAGMENT } from "queries/fragments";
+import CreateTransferShipment, {
+  IOrganisationsAgreementsDataData,
+  ITransferShipmentFormData,
+} from "./components/CreateTransferShipment";
+
+export const ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY = gql`
+  ${BASE_ORG_FIELDS_FRAGMENT}
+  ${TRANSFER_AGREEMENT_FIELDS_FRAGMENT}
+  query AllAcceptedTransferAgreements($baseId: ID!) {
+    base(id: $baseId) {
+      ...BaseOrgFields
+    }
+
+    transferAgreements(states: Accepted) {
+      ...TransferAgreementFields
+    }
+  }
+`;
+
+export const CREATE_SHIPMENT_MUTATION = gql`
+  mutation CreateTransferShipment(
+    $sourceBaseId: Int!
+    $targetBaseId: Int!
+    $transferAgreementId: Int!
+  ) {
+    createShipment(
+      creationInput: {
+        sourceBaseId: $sourceBaseId
+        targetBaseId: $targetBaseId
+        transferAgreementId: $transferAgreementId
+      }
+    ) {
+      id
+    }
+  }
+`;
+
+function CreateTransferShipmentView() {
+  // Basics
+  const navigate = useNavigate();
+  const { triggerError } = useErrorHandling();
+  const { createToast } = useNotification();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // variables in URL
+  const baseId = useParams<{ baseId: string }>().baseId!;
+
+  // Query Data for the Form
+  const allFormOptions = useQuery<AllAcceptedTransferAgreementsQuery>(
+    ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY,
+    {
+      variables: {
+        baseId,
+      },
+    },
+  );
+
+  // Mutation after form submission
+  const [createTransferShipmentMutation, createTransferShipmentMutationState] = useMutation<
+    CreateTransferShipmentMutation,
+    CreateTransferShipmentMutationVariables
+  >(CREATE_SHIPMENT_MUTATION);
+
+  // Prep data for Form
+  const allTransferAgreements = allFormOptions.data?.transferAgreements;
+
+  // eslint-disable-next-line max-len
+  const currentOrganisationLabel = `${allFormOptions?.data?.base?.organisation?.name} - ${allFormOptions?.data?.base?.name}`;
+
+  const partnerOrgsAgreementData = allTransferAgreements
+    ?.filter(
+      (agreement) =>
+        agreement.sourceOrganisation.id === globalPreferences.selectedOrganisationId?.toString() &&
+        (agreement.type === TransferAgreementType.SendingTo ||
+          agreement.type === TransferAgreementType.Bidirectional),
+    )
+    .map(
+      (agreement) =>
+        ({
+          agreementId: agreement.id,
+          specialNote: agreement.comment,
```

Comment:
This field, and orgId and orgName are not used elsewhere if I'm not mistaken. Can we remove them here?

Reply:
NVM I see that they're used in the low-level component.